### DISCUSSION TITLE
Software supply chain integration

### DISCUSSION THREAD

**Discussion by filintod** [fluxcd/flux2#464] [Proposals] [answered]
I think that as well as checking signed commits and or getting information from SOPs secrets, it would be nice to integrate with a software supply chain ([in-toto](https://in-toto.io/) or [grafeas](https://grafeas.io/)) before a deployment is executed. Is there any way to do it at the moment or is it in the future plans?

**Comment by kingdonb** [comment, marked as answer]
I think the way to support that is with a Validating Admission Webhook, I don't know if either software package includes that kind of thing.

You would really like to validate the images before they get sent to the production cluster, so maybe it would be a good idea to implement validating admission webhooks in staging as well as production? I do not think there is any special support in Flux for this workflow today.
